### PR TITLE
Sphinx Documentation, Readme, and Example Notebook.

### PR DESCRIPTION
**Pull request by KPrasch**
This PR aims to make pyUmbral more accessible; Introducing the sphinx configuration for auto-building documentation hosted on read the docs.

Documentation is seeded with the index, installation documentation, and one Jupyter Notebook demonstrating the usage of the "Simple API".

The project readme is also updated, containing overlapping information with the sphinx source code.

~ Feedback is welcome ~

### CODE REVIEW

**Comment by KPrasch** [comment]
:+1:

**Comment by KPrasch** [comment]
moved to PR #72

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+   :language: bash
+
+
+pyUmbral
+========
+
+.. image:: https://travis-ci.org/nucypher/pyUmbral.svg?branch=master
+    :target: https://travis-ci.org/nucypher/pyUmbral
+
+pyUmbral is a 100% python implementation of Umbral using OpenSSL and Cryptography.io,
```
Comment (tuxxy) [inline README.rst]: Actually, this isn't accurate. It uses C OpenSSL bindings w/ cryptography.io

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    # Bob receives the capsule through a side channel (S3, ipfs, Google Cloud, etc.)
+    capsule = capsule
+
+    # Generate split re-encryption keys with "M of N"
+    # Minimum threshold of 10 ("M") - and 20 total shares ("N").
+    kfrags, _ = umbral.split_rekey(alices_private_key, bobs_public_key, 10, 20)
+
+    # Ursula re-encrypts the shares and bob attaches them to the capsule
+    for kfrag in kfrags:
+        cfrag = umbral.reencrypt(kfrag, umbral_capsule)
```
Comment (tuxxy) [inline README.rst]: `umbral_capsule` isn't defined here.

Comment (KPrasch) [reply README.rst]: Ah, good catch.

File: README.rst (hunk 1 above)
Comment (KPrasch) [reply README.rst]: Right, I'll remove it.

Comment (KPrasch) [reply README.rst]: :heavy_check_mark:

File: README.rst (hunk 2 above)
Comment (KPrasch) [reply README.rst]: :heavy_check_mark:

**Comment by KPrasch** [comment]
Thanks for the reviews! :+1:  Keep em coming, it takes many pairs of eyes to get open source documentation right.

**Comment by KPrasch** [comment]
Merge with #72

**Pull request closed without merging**